Tests/NIOTests/RecvByteBufAllocatorTest.swift

```diff
@@ -113,12 +113,12 @@ final class AdaptiveRecvByteBufferAllocatorTest: XCTestCase {
         self.adaptive = AdaptiveRecvByteBufferAllocator(minimum: 0, initial: .max / 2, maximum: .max)
         var mayGrow = true
         while mayGrow {
-            mayGrow = self.adaptive.record(actualReadBytes: .max)
+            mayGrow = self.adaptive.record(actualReadBytes: .max / 2 )
```

Comment:
Why was this change necessary?

Reply:
With this set to `Int32.max`, [this line keeps doubling `nextReceiveBufferSize` until it overflows on 32-bit armv7](https://github.com/apple/swift-nio/blob/main/Sources/NIO/RecvByteBufferAllocator.swift#L95), then [this precondition trips because it thinks`nextReceiveBufferSize` is negative and therefore less than `minimum`](https://github.com/apple/swift-nio/blob/main/Sources/NIO/RecvByteBufferAllocator.swift#L85).

This test change is just working around that overflow, should probably fix that to not overflow on 32-bit platforms instead.

Reply:
That precondition shouldn't trip. Instead, the line underneath should be rewritten to be ` upperBound <= 0 ?` instead. That will generate the correct logic, which is that we'll clamp at the maximum size.

Reply:
Will do.

Reply:
Made that change to check if `upperBoundCandidate` is negative, but I still had to make this change to the test. Otherwise, the `nextReceiveBufferSize` correctly maxes out at 2^30 and this test loop becomes an infinite loop: I guess because it can never reach `Int32.max`.

Reply:
Oh, line 105 of `RecvByteBufferAllocator` needs an extra check: it needs to check that `self.nextReceiveBufferSize != upperBound`. Otherwise this isn't growing at all.